.github/workflows/ci.yml

```diff
@@ -91,11 +91,17 @@ jobs:
         run: npm run test:ui
 
       - name: Test end-to-end (Linux)
-        run: xvfb-run -a npm run test:e2e
         if: ${{ matrix.os == 'ubuntu-latest' }}
+        uses: nick-fields/retry@v3.0.0
+        with:
+          command: xvfb-run -a npm run test:e2e
+          max_attempts: 2
+          timeout_minutes: 10
       - name: Test end-to-end (Windows / macOS)
-        run: npm run test:e2e
         if: ${{ matrix.os != 'ubuntu-latest' }}
+        uses: Wandalen/wretry.action@v3.7.3
+        with:
+          command: npm run test:e2e
```

Comment:
This one is more flexible (it also allows you to retry actions themselves, not just commands) and has sensible defaults. But, it has less stars